src/services/expenses.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+let expenses = [];
+
+function init() {
+  expenses = [];
+};
+
+function getAll(userId, category, from, to) {
+  let currentExpenses = expenses;
+
+  if (userId) {
+    currentExpenses = currentExpenses
+      .filter(expense => expense.userId === +userId);
+  }
+
+  if (category) {
+    currentExpenses = currentExpenses
+      .filter(expense => expense.category === category);
+  }
+
+  if (from) {
+    currentExpenses = currentExpenses
+      .filter(expense => expense.spentAt >= from);
+  }
+
+  if (to) {
+    currentExpenses = currentExpenses
+      .filter(expense => expense.spentAt <= to);
+  }
+
+  return currentExpenses;
+};
+
+function getbyId(expenseId) {
+  const foundExpense = expenses.find(expense => expense.id === +expenseId);
+
+  return foundExpense || null;
+}
+
+function create(userId, spentAt, title, amount, category, note) {
+  const newExpense = {
+    id: expenses.length + 1,
```

Comment:
what if we delete one (not last) and then add?
there will be expenses with duplicated ids

Reply:
try to find max and then increment

Reply:
but don't care, in real life db will handle it instead of you